### ISSUE TITLE
[bug] icemix.py port: Optimizer not handling weight decay for cls_token

### ISSUE THREAD

**Issue by timinar** [graphnet-team/graphnet#713]
The `DeepIce` model contains a method called `no_weight_decay()` which is intended to specify that the `cls_token` parameter should not be subject to weight decay during training:
```
@torch.jit.ignore
def no_weight_decay(self) -> Set:
    """cls_tocken should not be subject to weight decay during training."""
    return {"cls_token"}
```
However, `optimizer_grouped_parameters` are not specified during training, so this method has no effect.
I believe that in the original 2nd place code, FastAI's wrapper around AdamW handled this automatically.

**Comment by ChenLi2049** [comment]
Yes, this method `def no_weight_decay()` is [called in BEiT-2 when training](https://github.com/microsoft/unilm/blob/master/beit/run_class_finetuning.py#L474). However, I searched for it and it is not called in [the original 2nd place solution](https://github.com/search?q=repo%3ADrHB%2Ficecube-2nd-place%20no_weight_decay&type=code) or [fastai](https://github.com/search?q=repo%3Afastai%2Ffastai%20no_weight_decay&type=code).

Also, I can find `optimizer_grouped_parameters` in BEiT-2, but not in the original 2nd place solution or fastai.

I find this to be a historical reason and maybe this method `def no_weight_decay()` can be removed.
